content/en/Features/db_migration.md

```diff
@@ -0,0 +1,257 @@
+as part of the deployment to a "staging/qa" environment.
+
+### Have Folks on Standby
+
+Always prepare under the assumption that if something can go wrong, it will go wrong. Hence, always have someone on 
+standby monitoring the migration.
+
+## DB Migration Options with Pure Opta
+Now that we have the preparations in place, there are several ways which database migrations can be done with Opta,
+depending on your needs/situation.
+
+### On a Server's Startup
+The simplest form is having the migration command run as part of server startup. This option is arguably the easiest/best
+to begin with and possible due to the aforementioned locking nature of migration, ensuring that even if multiple servers 
+are started simultaneously your data will not be tainted (assuming you use a tool which keeps track of the migration 
+version, which is most-if-not-all popular ones). For instance, consider the following startup script for a 
+[flask](https://flask.palletsprojects.com/en/2.0.x/) server which runs the migration and then starts the server.
+
+```shell
+set -exo pipefail
+
+export PYTHONPATH=$(pwd)
+
+source $(pipenv --venv)/bin/activate
+
+cd srv
+flask db upgrade # Does the migrations
+gunicorn -c gunicorn.py wsgi # Starts the server
+```
+
+As long as the migrations are short, this option should work fine, keep your database and server changes in lock step 
+and keep your ci/cd unchanged.
+
+### In a Running Server's Container
+An alternative route is having a user enter one of the running containers via the `opta shell` command and execute the 
+migration manually. This option offers more developer control, but now requires a human in the loop, does not enforce
+synchronicity between the database and server versions, and users will need to be careful not to trigger a new deployment
+while the operation is taking place.
+
+### In a One-Time Job
+For a cleaner separation of migration and server deployment, you can also run a migration as a separate 
+[kubernetes job](https://kubernetes.io/docs/concepts/workloads/controllers/job/) by using a helm chart module
+to create the job with the required inputs. This can be done in either the existing service's manifest, or a new 
+manifest which refers to the current manifest for data such as the current container image deploy (we recommend the latter
+approach). Take for example the following manifests which starts with a k8s service with a database and then uses a known 
+public helm chart which creates a one-time job for the migration in either the same manifest or a new one:
+
+{{< tabs tabTotal="2" >}}
+{{< tab tabName="New Manifest" >}}
+
+```yaml
+environments:
+  - name: staging
+    path: "./staging.yaml"
+name: my-service-migration
+modules:
+  - type: external-state
+    name: external
+    backend_type: s3
+    config:
+      bucket: opta-tf-state-my-org-staging
+      key: my-service
+      region: us-east-1
```

Comment:
why are we using `external-state` here? it's not related to this topic.
Specially since it's a new module, some users might not understand that this is not needed.

Reply:
`This can be done in either the existing service's manifest, or a new manifest which refers to the current manifest for data such as the current container image deploy`